### ISSUE TITLE
Showing CORS Issue instead of showing relevant information on 500 Errors as response

### ISSUE THREAD

**Issue by varun2948** [hotosm/fmtm#1353]
**Describe the bug**
Whenever we have an issue related with backend mostly when it's 500 status code it shows CORS Error when it's not actually a CORS issue.

**To Reproduce**

1. Go to create project
2. Fill the form, draw an AOI.
3. Generate data extract multiple times

**Expected behavior**
It should give the frontend the reasonable problem message.

**Comment by saram-aman** [comment]
@varun2948 can i work on this issue?

**Comment by varun2948** [comment]
@saram-aman sure, you can go ahead.

**Comment by spwoodcock** [comment]
Normally a response from the API would included a `Access-Control-Allow-Origin` header, informing the browser if the request violates CORS.

When we get a 500 error on the API, the request is effectively terminated, so no response is returned. This means that no `Access-Control-Allow-Origin` is included, so the browser can't determine if there is a CORS violation (i.e. the CORS error is triggered).

There is a workaround for this for FastAPI: we could make a middleware that catches request exceptions, and intercept them, instead returning a response with a 500 response.

However, a 500 response doesn't really indicate anything other than 'There was an error' (we can't get more info, as it's an **unhandled** exception), so I think a message like that is adequate on the frontend side.

tl;dr if the request fails on the frontend, we should show a generic error 'There was an error'.
